nd4j/nd4j-common/src/main/java/org/nd4j/resources/Downloader.java

```diff
@@ -95,6 +95,8 @@ private static void downloadAndExtract(int attempt, int maxTries, String name, U
             }
             // try extracting
             try{
+                if (!extractToDir.exists())
```

Comment:
Can you add a test for this?

Reply:
Done. And actually moved this code to the the ArchiveUtils.unzipFileTo since that would be the more correct place for it.